### PR TITLE
changes to encoder to update to Symfony

### LINKED ISSUE #9831: Mautic 4: Unable to edit user account
⚠️ This relates to Mautic 4.0-alpha1

## Bug Description
When editing the logged in user profile (I added a position) a 500 error is thrown.

| Q   | A
| --- | ---
| Mautic version | 4.0-alpha1
| PHP version | 7.4
| Browser | Chrome

### Steps to reproduce
1. Log into Mautic
2. Edit your user profile
2. Save
 
### Log errors
```
[2021-03-25 18:38:05] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: "The "security.encoder_factory" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead." at /var/www/html/vendor/symfony/dependency-injection/Container.php line 275 {"exception":"[object] (Symfony\\Component\\DependencyInjection\\Exception\\ServiceNotFoundException(code: 0): The \"security.encoder_factory\" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead. at /var/www/html/vendor/symfony/dependency-injection/Container.php:275)
[stacktrace]
#0 /var/www/html/vendor/symfony/dependency-injection/Container.php(231): Symfony\\Component\\DependencyInjection\\Container->make('security.encode...', 1)
#1 /var/www/html/vendor/symfony/framework-bundle/Controller/ControllerTrait.php(68): Symfony\\Component\\DependencyInjection\\Container->get('security.encode...')
#2 /var/www/html/app/bundles/UserBundle/Controller/ProfileController.php(156): Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller->get('security.encode...')
#3 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(158): Mautic\\UserBundle\\Controller\\ProfileController->indexAction()
#4 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(80): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)
#5 /var/www/html/vendor/symfony/http-kernel/Kernel.php(201): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#6 /var/www/html/app/AppKernel.php(104): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#7 /var/www/html/app/middlewares/CORSMiddleware.php(91): AppKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#8 /var/www/html/app/middlewares/CatchExceptionMiddleware.php(43): Mautic\\Middleware\\CORSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#9 /var/www/html/app/middlewares/Dev/IpRestrictMiddleware.php(61): Mautic\\Middleware\\CatchExceptionMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#10 /var/www/html/app/middlewares/VersionCheckMiddleware.php(67): Mautic\\Middleware\\Dev\\IpRestrictMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#11 /var/www/html/app/middlewares/TrustMiddleware.php(51): Mautic\\Middleware\\VersionCheckMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#12 /var/www/html/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Mautic\\Middleware\\TrustMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#13 /var/www/html/vendor/stack/run/src/Stack/run.php(13): Stack\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))
#14 /var/www/html/index_dev.php(25): Stack\
un(Object(Stack\\StackedHttpKernel))
#15 {main}
"} 
```

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #9831

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Try to create a user
3. Update a user
4. There should be testes for API and SAML as well

### CODE REVIEW

**Comment by mabumusa1** [comment]
For reference this article included the changes required http://securepasswords.info/symfony/

**Review by RCheesley — APPROVED** [review]
Bug reproduced and the PR applied fixes the issue - was able to create a new user and to log in as that user with the correct access rights! Thanks @mabumusa1 🚀

File: app/bundles/InstallBundle/Controller/InstallController.php [hunk 1]
```diff
@@ -21,7 +21,7 @@
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
-use Symfony\Component\Security\Core\Encoder\PasswordEncoderInterface;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoder;
```
Comment (escopecz) [inline app/bundles/InstallBundle/Controller/InstallController.php:24]: This use statement is not used anywhere and can be removed. This is just a nitpick, nothing that should block the PR merge.

**Review by escopecz — APPROVED** [review]
I see no issue in the code. Also tested changing password via the profile form and log in.  👍

**Pull request merged**